src/algebra/order/hom/ring.lean

```diff
@@ -294,5 +298,42 @@ def to_order_ring_hom (f : α ≃+*o β) : α →+*o β :=
 @[simp]
 lemma coe_to_order_ring_hom_refl : (order_ring_iso.refl α : α →+*o α) = order_ring_hom.id α := rfl
 
+lemma to_order_ring_hom_injective : injective (to_order_ring_hom : (α ≃+*o β) → α →+*o β) :=
+λ f g h, fun_like.coe_injective $ by convert fun_like.ext'_iff.1 h
+
 end non_assoc_semiring
 end order_ring_iso
+
+/-!
+### Unicity
+
+There is at most one ordered ring homomorphism from a linear ordered field to an archimedean linear
+ordered field. Reciprocally, such an ordered ring homomorphism exists when the codomain is further
+conditionally complete.
+-/
+
+/-- There is at most one ring homomorphism from a linear ordered field to an archimedean linear
+ordered field. -/
+instance [linear_ordered_field α] [linear_ordered_field β] [archimedean β] :
+  subsingleton (α →+*o β) :=
```

Comment:
Considering #6025, and the fact that these instances are rarely going to be used, I think these should be non-instances.

Reply:
You should add an exact copy of the comment elsewhere we have referencing 6025, so that we can eventually remember to make this an instance